nucypher/blockchain/eth/deployers.py

```diff
@@ -423,11 +554,57 @@ def get_latest_version(cls, blockchain) -> Contract:
                                                              proxy_name=cls.__proxy_deployer.contract_name)
         return contract
 
+    def upgrade(self, existing_secret_plaintext: bytes, new_secret_hash: bytes):
+
+        deployment_transactions = dict()
+
+        existing_bare_contract = self.blockchain.interface.get_contract_by_name(name=self.contract_name,
+                                                                                proxy_name=self.__proxy_deployer.contract_name,
+                                                                                use_proxy_address=False)
+        # Proxy-Proxy
+        proxy_deployer = self.__proxy_deployer(deployer_address=self.deployer_address,
+                                               target_contract=existing_bare_contract,
+                                               bare=True)
+
+        # Proxy
+        proxy_args = (self.contract_name,
+                      self.token_agent.contract_address,
+                      self.miner_agent.contract_address,
+                      self.policy_agent.contract_address)
+
+        user_escrow_proxy_contract, proxy_deployment_txhash = self.blockchain.interface.deploy_contract(*proxy_args)
+        self._contract = user_escrow_proxy_contract
+        deployment_transactions['deployment_txhash'] = proxy_deployment_txhash
+
+        proxy_deployer.retarget(new_target=user_escrow_proxy_contract.address,
+                                existing_secret_plaintext=existing_secret_plaintext,
+                                new_secret_hash=new_secret_hash)
+
+        deployment_transactions['proxy_deployment'] = proxy_deployment_txhash
+
+        return deployment_transactions
+
+    def rollback(self, existing_secret_plaintext: bytes, new_secret_hash: bytes):
```

Comment:
There is no lowlevel `rollback` function for `UserEscrowProxy`. If we want to rollback - then upgrade to previous address

Reply:
:heavy_check_mark: